juju/juju
# [JUJU-4012] Wire up watcher registry

The following wires up the watcher registry to the pinger timeout watcher.
This is the first watcher to take advantage of the watcher registry. We're
deprecating the `Resource` type in favour of a real `worker.Worker`, that
will correctly monitor the lifecycle of a watcher.

This supersedes https://github.com/juju/juju/pull/15767 which required too
much churn. Instead, it will require every person to migrate each watcher
over to the `WatcherRegistry` as we move along. This is the cleanest 
bulletproof way to ensure that we don't break anything as we go.

As a driveby, I cleaned up the API server to make it readable.

## Checklist


## QA steps

Ensure there are no errors when adding and removing models.

```sh
$ juju bootstrap lxd test --build-agent
$ juju debug-log -m controller
$ juju add-model default
$ juju destroy-model default
```
